.travis.yml

```diff
@@ -80,31 +83,34 @@ jobs:
         # preload simulator
         - Scripts/start_simulator.sh
       script:
-        - if [[ "$TRAVIS_BRANCH" == "master" ]]; then xcodebuild test -workspace OptimizelySwiftSDK.xcworkspace -scheme $SCHEME -configuration Release CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO -sdk $TEST_SDK -destination "platform=$PLATFORM,OS=$OS,name=$NAME" ONLY_ACTIVE_ARCH=YES | tee buildoutput | xcpretty && test ${PIPESTATUS[0]} -eq 0; fi
+        - if [[ "$TRAVIS_BRANCH" == "master" ]]; then xcodebuild test -derivedDataPath $COVERAGE_DIR  -workspace OptimizelySwiftSDK.xcworkspace -scheme $SCHEME -configuration Release CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO -sdk $TEST_SDK -destination "platform=$PLATFORM,OS=$OS,name=$NAME" ONLY_ACTIVE_ARCH=YES | tee buildoutput | xcpretty && test ${PIPESTATUS[0]} -eq 0; fi
       after_success:
-        # coverage collect for iOS only (.slather.yml)
-        - if [[ "$PLATFORM" == "iOS Simulator" ]]; then slather; fi
+        # [coveralls]
+        # - exclude coverage for Test codes by setting OptimizelySwiftSDK-iOS scheme > Test > Options > Gather coverage for selected targets
+        # - report coverage for PR and iPhone 11 only (avoid redundant ones)
+        # - use Xcode12.4+ (older Xcode reports a wrong number)
+        - if [[ "$TRAVIS_BRANCH" == "master" && "$PLATFORM" == "iOS Simulator" && "$NAME" == "iPhone 11" ]]; then Scripts/upload_coveralls.sh; fi
```

Comment:
can we move line# 86  to a single script file. Will be more cleaner in travis file.